sys/net/network_layer/sixlowpan/icmp.c

```diff
@@ -1183,7 +1183,7 @@ void recv_nbr_sol(void)
         /* solicited na */
         uint8_t flags = (ICMPV6_NEIGHBOR_ADV_FLAG_OVERRIDE | ICMPV6_NEIGHBOR_ADV_FLAG_SOLICITED);
         icmpv6_send_neighbor_adv(&(ipv6_buf->srcaddr), &(ipv6_buf->destaddr),
-                                 alist_targ.addr->addr_data, flags, 0, OPT_ARO);
+                                 alist_targ.addr->addr_data, flags, OPT_TLLAO, 0);
```

Comment:
I believe this should be split into two branches, one for NS which contained a valid ARO option, one for replies to non-ARO NS, since the behaviour differ between the two.
For ARO solicitations, the reply `SHOULD` include a copy of the ARO with status set to zero according to RFC 6775, and does not need a TLLAO.
For non-ARO solicitations, it makes no sense to include a broken ARO, and a TLLAO is mandatory.

https://tools.ietf.org/html/rfc6775#section-6.5.3